tests/testthat/test-build_request.R

```diff
@@ -0,0 +1,95 @@
+  expect_identical(req$endpoint, "drive.files.list")
+  expect_identical(req$verb, "GET")
+  expect_identical(req$url, "https://www.googleapis.com/drive/v3/files")
+
+})
+
+test_that("build_request messages effectively with input", {
+  ## the only piece that is "user facing" is you can supply
+  ## parameters to the ...
+
+  ## if we give a crazy parameter that isn't found in our
+  ## master tibble, housed in .drive$params, we should error
+  params <- list(chicken = "muffin")
+  expect_message(build_request(params = params),
+                 paste(
+                   c(
+                     "Ignoring these unrecognized parameters:",
+                     glue::glue_data(
+                       tibble::enframe(params),
+                       "{name}: {value}"
+                     )
+                   ),
+                   collapse = "\n")
+  )
+
+  ## and if we supply multiple crazy parameters
+  params <- list(chicken = "muffin",
+                 bunny = "pippin")
+  expect_message(build_request(params = params),
+                 paste(
+                   c(
+                     "Ignoring these unrecognized parameters:",
+                     glue::glue_data(
+                       tibble::enframe(params),
+                       "{name}: {value}"
+                     )
+                   ),
+                   collapse = "\n")
+  )
+
+  ## what if some are right and some are wrong?
+
+  params <- list(chicken = "muffin",
+                 q = "fields='files/id'")
+  params_wrong <- params[1]
+  params_right <- params[2]
+  expect_message(build_request(params = params),
+                 paste(
+                   c(
+                     "Ignoring these unrecognized parameters:",
+                     glue::glue_data(
+                       tibble::enframe(params_wrong),
+                       "{name}: {value}"
+                     )
+                   ),
+                   collapse = "\n")
+  )
+
+  ## let's make sure the correct parameter (q) still passed
+  expect_identical(suppressMessages(build_request(params = params)$query),
+                   params_right)
+
+  ## what if you give a parameter that is accepted in a
+  ## different endpoint, but not the one you specified?
+  ## should still give a message
+
+  ## for the permissions, must supply fileId
+  params <- list(q = "this is not the q you're looking for",
+                 fileId = 1)
+
+  ## q is not an allowed parameter for listing permissions
+  params_wrong <- params[1]
+
+  expect_message(build_request(endpoint = "drive.permissions.list",
+                               params = params),
+                 paste(
+                   c(
+                     "Ignoring these unrecognized parameters:",
+                     glue::glue_data(
+                       tibble::enframe(params_wrong),
+                       "{name}: {value}"
+                     )
+                   ),
+                   collapse = "\n")
+  )
+
+
+})
```

Comment:
These look good. My only advice would be, again, to split into smaller tests with pretty tight goals, if you can.